lib/data_access/http.service.ts

```diff
@@ -45,7 +45,7 @@ export class Http{
 				(<any>currentHttpConfig.headers)["Content-Type"] = "application/json";
 		}
 
-		return ajax(Object.assign({
+		return (ajaxService || ajax)(Object.assign({
```

Comment:
To support libraries that return Promises, you'll probably want to use `ajaxService ? from(ajaxService(...)) : ajax(...)`, so your ajax implementation doesn't have to know rxjs.

Reply:
Ajax implementation needs to know rxjs, as we pass the `AjaxRequest` object to it.